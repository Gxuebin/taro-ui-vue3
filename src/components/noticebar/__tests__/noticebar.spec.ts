import { mount } from '@vue/test-utils'
import AtNoticebar from '../index'

const factory = (values = {}, slots = { default: ['这是内容'] }) => {
  return mount(AtNoticebar as any, {
    slots,
    props: { ...values },
  })
}

describe('AtNoticebar', () => {
  it('should render prop -- show', () => {
    const wrapper = factory({ close: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- single', () => {
    const wrapper = factory({ single: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- speed', () => {
    const wrapper = factory({ speed: 200 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- moreText & showMore', () => {
    // showMore work only when single is true
    const wrapper = factory({ moreText: '查看更多', showMore: true, single: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- icon', () => {
    const wrapper = factory({ icon: 'volume-plus' })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('AtNoticebar Behavior', () => {
  it('should trigger onClose', () => {
    const onClose = jest.fn()
    const wrapper = factory({
      close: true,
      onClose: onClose,
    })
    wrapper.find('.at-noticebar__close').trigger('tap')
    expect(onClose).toBeCalled()
  })

  it('should trigger onGotoMore', () => {
    const onGotoMore = jest.fn()
    const wrapper = factory({
      icon: 'volume-plus',
      single: true,
      showMore: true,
      onGotoMore: onGotoMore,
      moreText: '更多内容',
    })
    wrapper.find('.at-noticebar__more').trigger('tap')
    expect(onGotoMore).toBeCalled()
  })
})
