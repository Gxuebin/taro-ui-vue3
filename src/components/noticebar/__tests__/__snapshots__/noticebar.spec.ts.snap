// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtNoticebar should render close icon and match snapshot 1`] = `
<view
  class="at-noticebar__close"
>
  <text
    class="at-icon at-icon-close"
  />
</view>
`;

exports[`AtNoticebar should render moreText in single-line mode and match snapshot 1`] = `
<view
  class="at-noticebar__more"
>
  <text
    class="text"
  >
    查看更多
  </text>
  <view
    class="at-noticebar__more-icon"
  >
    <text
      class="at-icon at-icon-chevron-right"
    />
  </view>
</view>
`;

exports[`AtNoticebar should render prop icon and match snapshot 1`] = `
<view
  class="at-noticebar__content-icon"
>
  <text
    class="at-icon at-icon-volume-plus"
  />
</view>
`;

exports[`AtNoticebar should render prop speed 1`] = `
<view
  animation="[object Object]"
  class="at-noticebar__content-inner J_kue4"
  id="J_kue4"
  style="animation-duration: 15s;"
>
  this is a slot content
</view>
`;

exports[`AtNoticebar should render slot content and match snapshot 1`] = `
<view
  animation="[object Object]"
  class="at-noticebar__content-inner"
  id="J_fy4i"
>
  this is slot content text
</view>
`;
